### ISSUE TITLE
Filter + fillna breaks

### ISSUE THREAD

**Issue by xdssio** [vaexio/vaex#242]
```
import vaex
data = vaex.from_csv('titanic.csv')
train, test = data.split_random(0.5, random_state=42)
train['SibSp'] = train['SibSp'].fillna(0)
train = train[train.SibSp==1]
train.head()


 to the filtering, the (unfiltered) length is %s." % (len(ar), len(self), self.length_unfiltered()))
-> 2628                 raise ValueError("array is of length %s, while the length of the DataFrame is %s" % (len(ar), self.length_original()))
   2629             # assert self.length_unfiltered() == len(data), "columns should be of equal length, length should be %d, while it is %d" % ( self.length_unfiltered(), len(data))
   2630             self.columns[name] = f_or_array

ValueError: array is of length 446, while the length of the DataFrame is 113
```

**Comment by maartenbreddels** [comment]
Hi J,

Thanks for the report. I could reproduce it, using the titanic dataset of kaggle (please mention that next time or reduce the issue to a self-contained example).

cheers,

Maarten

**Comment by maartenbreddels** [comment]
Fixed in #231
